src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java

```diff
@@ -244,38 +251,31 @@ public AnalysisFile saveFileByLink(String link, IUser user, A analysis, String l
 
     @PreAuthorize("hasPermission(#analysis, "
             + "T(com.odysseusinc.arachne.portal.security.ArachnePermission).UPLOAD_ANALYSIS_FILES)")
-    public void saveCOHORTAnalysisArchive(A analysis, DataReference dataReference, IUser user, List<MultipartFile> files) {
-
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
-        StringBuilder generatedFileNameBuilder = new StringBuilder();
-        generatedFileNameBuilder.append(CommonAnalysisType.COHORT.getTitle());
-        try (final ZipOutputStream zos = new ZipOutputStream(out)) {
-            files.stream()
+    public void saveCohortAnalysisArchive(A analysis, DataReference dataReference, IUser user, List<MultipartFile> files) {
+
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream();
+             ZipOutputStream zos = new ZipOutputStream(out)) {
+            MultipartFile genericSqlFile = files.stream()
```

Comment:
Now the whole collection handled with two loops. Not sure this is better than single-phase handling.

Reply:
I rewrote it in a more readable way(at least I think so)  I don't really like it when we iterate over the collection and do more that one thing at the time.